### PR TITLE
Array helper libs

### PR DESCRIPTION
**Pull request by fedokDL**

### CODE REVIEW

File: contracts/libs/ArrayHelper.sol [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        reversed = new uint256[](arr.length);
+
+        uint256 lastIndex = arr.length - 1;
+        for (uint256 i; i < arr.length; i++) {
+            reversed[i] = arr[lastIndex - i];
+        }
+    }
+
+    function part(
+        uint256[] storage arr,
```
Comment (Hrom131) [inline contracts/libs/ArrayHelper.sol:20]: Why we should pass storage variable?

File: contracts/libs/ArrayHelperBase.sol [hunk 2]
```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import "./ArrayHelperBase.sol";
```
Comment (Hrom131) [inline contracts/libs/ArrayHelperBase.sol:4]: Useless import

**Comment by Hrom131** [comment]
I'm not sure that `ArrayHelperBase` and `ArrayHelper` are appropriate names for libraries of this kind.

It seems to me that it would be better to create one library `PaginationHelper`, which would have functionality of ...Base library, as well as overloaded functions for different types (`uint256`, `address`, `bytes32`, `AddressSet`, etc.)

In the `ArrayHelper` library is well suited function `reverse`, you can also add a function `slice`, which will take parts of the array

**Comment by Arvolear** [comment]
1. Please move all the arrays related libs into the "arrays" folder inside the libs directory (the same with mocks)
2. Let's not include the assembly paginator in the modules for now because it might be very error-prone

File: contracts/libs/arrays/ArrayHelper.sol [hunk 3]
```diff
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+library ArrayHelper {
```
Comment (Arvolear) [inline contracts/libs/arrays/ArrayHelper.sol:4]: Please add natspec description

**Pull request merged**